subsys/net/lib/dns/dns_cache.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2024 Endress+Hauser AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/net/dns_resolve.h>
+#include "dns_cache.h"
+
+LOG_MODULE_REGISTER(net_dns_cache, 4);
+
+static void dns_cache_clean(struct dns_cache const *cache);
+
+int dns_cache_flush(struct dns_cache *cache)
+{
+	k_mutex_lock(cache->mut, K_FOREVER);
+	for (size_t i = 0; i < cache->size; i++) {
+		cache->entries[i].in_use = false;
+	}
+	k_mutex_unlock(cache->mut);
+
+	return 0;
+}
+
+int dns_cache_add(struct dns_cache *cache, char const *query, struct dns_addrinfo const *addrinfo,
+		  uint32_t ttl)
+{
+	int64_t oldest_entry_uptime = INT64_MAX;
+	size_t index_to_replace = 0;
+	bool found_empty = false;
+
+	if (cache == NULL || query == NULL || addrinfo == NULL || ttl == 0) {
+		return -EINVAL;
+	}
+
+	if (strlen(query) >= DNS_MAX_NAME_LEN) {
+		LOG_WRN("Query string to big to be processed %u >= DNS_MAX_NAME_LEN",
```

Comment:
I replaced the normal log macros. A question out of interest, what is the difference? Both only show when NET_LOG is activated since `CONFIG_DNS_RESOLVER_LOG_LEVEL` only gets set when this is the case. So I am wondering if there is another difference. Sorry if this might be a stupid question.

Reply:
The NET_LOG macros are mostly same as normal logging macros, but for example NET_DBG prints the thread name or id which normal logging macro does not by default. The idea with separate networking logging macros is that we can adjust the printouts more easily for network specific stuff if needed.